src/Components/Blazor/DevServer/src/Server/Program.cs

```diff
@@ -18,12 +25,24 @@ public class Program
         /// <summary>
         /// Intended for framework test use only.
         /// </summary>
-        public static IWebHost BuildWebHost(string[] args) =>
-            WebHost.CreateDefaultBuilder(args)
-                .UseConfiguration(new ConfigurationBuilder()
-                    .AddCommandLine(args)
-                    .Build())
-                .UseStartup<Startup>()
-                .Build();
+        public static IHost BuildWebHost(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureHostConfiguration(cb => {
+                    var applicationPath = args.SkipWhile(a => a != "--applicationpath").Skip(1).FirstOrDefault();
+                    var name = Path.ChangeExtension(applicationPath,".StaticWebAssets.xml");
+
+                    if (name != null)
+                    {
+                        cb.AddInMemoryCollection(new Dictionary<string, string>
+                        {
+                            [WebHostDefaults.StaticWebAssetsKey] = name
```

Comment:
@Tratcher This is where the new API gets used.

Blazor standalone apps run using the blazor dev-server so it needs to be able to point to the manifest without using the client assembly as it won't be in its dependency graph

Reply:
Move the mitigation to launchsettings?

Reply:
We don't want that, as the whole point of this package is to setup a suitable host for your blazor standalone app (which is only files on disk).

BTW, don't worry about this a lot because this piece is not shipping for 3.0, but we want to fix this as customers have been complaining.